File: 30Jan2024.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Stack;

class Solution {
    public int evalRPN(String[] tokens) {
        Stack<String> stk = new Stack<>();
        HashSet<String> set = new HashSet<>(Arrays.asList(new String[] { "+", "-", "*", "/" }));
        // System.out.println(set);
        for (String t : tokens) {
            if (set.contains(t)) {
                int a = Integer.parseInt(stk.pop());
                int b = Integer.parseInt(stk.pop());
                stk.push(f(b, a, t.charAt(0)) + "");
            } else {
                stk.push(t);
            }
            // System.out.println(stk);
        }
        // System.out.println(stk);
        return Integer.parseInt(stk.pop());
    }

    public int f(int a, int b, char t) {
        if (t == '+') {
            return a + b;
        } else if (t == '-') {
            return a - b;
        } else if (t == '*') {
            return a * b;
        } else {
            return a / b;
        }
    }
}

File: 30Jan2024.py
from ast import List


class Solution:
    def f(self,b:int,a:int,t:chr) :
        if(t=='+') :
            return a+b
        elif(t=='-') :
            return a-b
        elif(t=='*') :
            return a*b
        else :
            return a/b
        
    def evalRPN(self, tokens: List[str]) -> int:
        st = ("-","+","*","/")
        stk = []
        for t in tokens :
            if(t in st) :
                stk.append(self.f(int(stk.pop()),int(stk.pop()),t[0]))
            else :
                stk.append(t)
        return int(stk[0])